### PR TITLE
[release] Update ray version 2.7.1

### PR DESCRIPTION
**Pull request by GeneDer**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/lint.rayci.yml [hunk 1]
```diff
@@ -73,4 +73,3 @@ steps:
       - pip install -r ci/compile_py_proto/requirements_compile_py_proto.txt
       - ./ci/lint/check-python-proto.sh
     depends_on: forge
-
```
Comment (aslonnie) [inline .buildkite/lint.rayci.yml:76]: why is this file change required?

File: release/ray_release/tests/test_wheels.py [hunk 2]
```diff
@@ -51,20 +51,20 @@ def test_get_ray_wheels_url(remove_buildkite_env):
         repo_url="https://github.com/ray-project/ray.git",
         branch="master",
         commit="1234",
-        ray_version="2.7.0",
+        ray_version="2.7.1",
```
Comment (aslonnie) [inline release/ray_release/tests/test_wheels.py:54]: I think you do not need to change these test files.

**Review by aslonnie — COMMENTED** [review]
what are those rllib tune metadata file changes?

File: .buildkite/lint.rayci.yml (hunk 1 above)
Comment (GeneDer) [reply .buildkite/lint.rayci.yml:76]: I don't think it's required, probably bc my ide was setup to drop extra time in the file. I can but the extra line back

**Comment by aslonnie** [comment]
> what are those rllib tune metadata file changes?

please also check those files.

they are binary files, and probably should not be changed.

also not sure also if they should be checked in


----

@can-anyscale , I think we should have a limited whitelist of places that require changes, not via repo-wide text replace.

**Comment by GeneDer** [comment]
> > what are those rllib tune metadata file changes?
> 
> please also check those files.
> 
> they are binary files, and probably should not be changed.
> 
> also not sure also if they should be checked in
> 
> @can-anyscale , I think we should have a limited whitelist of places that require changes, not via repo-wide text replace.

Interesting point, I think in the pervious process we would also change those file https://github.com/ray-project/ray/commit/2441e73d94a860c4f0385db94c31f3f09b2c73cf. But I do agree those shouldn't impact how things are working. Will revert the change and update the docs

**Review by aslonnie — APPROVED** [review]

**Pull request merged**